apps/native-component-list/src/navigation/MainTabNavigator.tsx

```diff
@@ -52,21 +57,28 @@ export default function MainTabbedNavigator(props: any) {
           borderTopWidth: StyleSheet.hairlineWidth,
           borderTopColor: Colors.tabIconDefault,
         }}
+        // bottom-tabs props
         tabBarOptions={{
           style: {
             backgroundColor: Colors.tabBar,
           },
           activeTintColor: Colors.tabIconSelected,
           inactiveTintColor: Colors.tabIconDefault,
         }}>
-        {Object.keys(Screens).map(name => (
-          <Tab.Screen
-            name={name}
-            key={name}
-            component={Screens[name].navigator}
-            options={Screens[name].navigator.navigationOptions}
-          />
-        ))}
+        {Object.keys(Screens).map(name => {
+          const screen = Screens[name];
+          if (screen) {
+            return (
+              <Tab.Screen
+                name={name}
+                key={name}
+                component={screen.navigator}
+                options={screen.navigator.navigationOptions}
+              />
+            );
+          }
+          return null;
+        })}
```

Comment:
Well, it is not a nicely written code 😞 
```suggestion
        {Object.entries(Screens)
           .filter(([name, Screen]) => Boolean(Screen))
           .map(([name, Screen]) => (
              <Tab.Screen
                name={name}
                key={name}
                component={screen.navigator}
                options={screen.navigator.navigationOptions}
              />
           ))
        }
```
Why some `Screens` not defined?
I see these are imported from `./MainNavigators` and the object exported there is of some explicitly typed structure. I didn't see any `undefined`s as values there 🤔

Reply:
I wanted to make as small changes as possible (keeping `keys.map` as it was before), but then TS complained about the fact that `Screens[name]` could be `undefined` (as per `Record<string, undefined | …>`), so I did the `if ()` so it knows it's not `undefined`.

Your proposed solution doesn't give TS confidence that `screen` is not `undefined`, there's an error.

Maybe the problem lies in the `undefined` that it should not be there (logically I thought it should be there but I may be wrong). 🤷‍♂️